app/models/job.ts

```diff
@@ -237,14 +237,29 @@ export class Job extends Record implements JobRecord {
 
   /**
   * Returns a Job with the given jobID using the given transaction
+  * Optionally locks the row.
   *
   * @param transaction - the transaction to use for querying
   * @param jobID - the jobID for the job that should be retrieved
+  * @param getLinks - if true include the job links when returning the job
+  * @param lock - if true lock the row in the jobs table
   * @returns the Job with the given JobID or null if not found
   */
-  static async byJobID(transaction: Transaction, jobID: string): Promise<Job | null> {
-    const jobList = await this.queryAll(transaction, { where: { jobID } }, true, 0, 1);
-    return jobList.data.shift();
+  static async byJobID(
+    transaction: Transaction, jobID: string, getLinks = true, lock = false,
+  ): Promise<Job | null> {
+
+    let query = transaction('jobs').select().where({ jobID });
+    if (lock) {
```

Comment:
This is really the key change for the PR - Lock the row in the jobs table when necessary. We will set lock to true as part of updating the job status in a work item update.